src/main/java/dk/kb/cdx/workflow/CdxIndexWorker.java

```diff
@@ -0,0 +1,131 @@
+        this.threadNumber=threadNumber;
+        this.cdxFormatBuilder = cdxFormatBuilder;
+        this.cdxServerUrl=cdxServerUrl;
+        this.absolutePath=absolutePath;
+        this.dryRun=dryRun;
+    }    
+    
+    
+
+    public WorkerStatus call() {
+        log.info("Starting CdxIndexerWorkerThread:"+threadNumber);                        
+
+        String nextWarcFile;
+        while( (nextWarcFile = CdxIndexerWorkflow.getNextWarcFile()) != null ) {    
+            try{
+                String cdxOutput=getCdxOutput(nextWarcFile, cdxFormatBuilder); //Exceptions are acceptable, can be corrupt WARC-files.
+                String responseBody=null;
+                try {
+                   responseBody=postCdxToServer(cdxServerUrl, cdxOutput); //Critital this does not fail. Stop thread instead of continue with something that can fail again and again
+                }
+                catch(Exception e) { //stop thread if CDX server is not running as expected.                         
+                 log.error("Stopping worker:"+threadNumber + " Error connecting to CDX server:"+e.getMessage() + " when process WARC file:"+nextWarcFile);  
+                 status.increaseErrors();
+                 return status; //Stop worker                     
+                }
+               if (responseBody != null) {
+                   responseBody= responseBody.trim(); //Remove a new line from the server as last character
+               }                
+                status.increaseCompleted();
+                log.info("Indexed:"+nextWarcFile +" result:"+responseBody);                 
+                CdxIndexerWorkflow.markWarcFileCompleted(nextWarcFile);
+            }
+            catch(Exception e){
+                status.increaseErrors();
+                log.error("Error processing WARC-file:"+nextWarcFile,e);
+                try {
+                    CdxIndexerWorkflow.markWarcFileCompleted(nextWarcFile);
+                }
+                catch(Exception eIO) {                        
+                   log.error("Error marking WARC file as completed. Stopping thread. WarcFile:"+nextWarcFile,eIO); // Has not happened yet
+
+                   return status; //Stop workflow  
+                }
+            }
+                        
+        }
+        log.info("Worker completed. No more WARC-files to process for CdxIndexerWorkerThread:"+threadNumber + ". Number processed:"+status.getCompleted() +" Number of errors:"+status.getErrors());                        
+        return status;
+    }
+    
+    /*
+     * Return the body message from the CDX server. If everything is well it will be something like: 'Added 80918 records'
+     * Will log error if HTTP status is not 200
+     */
+    private  String postCdxToServer(String cdxServer, String data) throws IOException,InterruptedException {     
+
+        if (dryRun) {
+         return "Added 0 records (dry run)";    
```

Comment:
Bit of a poor message. Maybe use `data.split("\n").length` for counting?

Reply:
good idea.